go/libraries/doltcore/schema/alterschema/droppk_test.go

```diff
@@ -124,3 +124,134 @@ func TestDropPk(t *testing.T) {
 		assert.True(t, ok)
 	})
 }
+
+func TestDropPks(t *testing.T) {
+	var dropTests = []struct {
+		name      string
+		setup     []string
+		exit      int
+		fkIdxName string
+	}{
+		{
+			name: "no error on drop pk",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id))",
+				"insert into parent values (1,1,1),(2,2,2)",
+			},
+			exit: 0,
+		},
+		{
+			name: "no error if backup index",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id), key `backup` (id))",
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if backup index for single FK on compound pk drop",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `backup` (age))",
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (age) references parent (age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if compound backup index for compound FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `backup` (id, age))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id, age) references parent (id, age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if both invalid and valid backup index",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `bad_backup1` (id, age), key `bad_backup2` (id), key `backup` (age, id))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (age) references parent (age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
```

Comment:
This is a good test, but I'd also add a case where there's just the single compound index and the foreign key only references the first key.